src/cascadia/TerminalApp/AppCommandlineArgs.cpp

```diff
@@ -285,42 +310,89 @@ void AppCommandlineArgs::_addNewTerminalArgs(CLI::App* subcommand)
+    // pass to the NewTerminalArgs. However, if that commandline contains any
+    // options that _we_ understand, CLI11 will try to first grab those options
+    // from the provided string to parse our own args, and _not_ include them in
+    // the _commandline here. Consider for example, the following command:
+    //
+    //     wt.exe new-tab wsl.exe -d Alpine
+    //
+    // We want the commandline here to be "wsl.exe -d Alpine zsh". However, by
+    // default, that'll be parsed by CLI11 as a _startingDirectory of "Alpine",
+    // with a commandline of "wsl.exe zsh".
+    //
+    // So, instead, when we see that there's a commandlineOption that's been
+    // parsed, we'll use that as our cue to handle the commandline ourselves.
+    //  * We'll start by going through all the options that were parsed by
+    //    CLI11, and clearing all of the ones after the first commandline
+    //    option. This will prevent use from acting on their values later.
+    //  * Then, we'll grab all the strings starting with the first commandline
+    //    string, and add them to the commandline for the NewTerminalArgs.
+    if (*subcommand.commandlineOption)
     {
-        std::string buffer;
-        size_t i = 0;
-        for (auto arg : _commandline)
+        auto opts = subcommand.subcommand->parse_order(); // const std::vector<Option*>&
+        auto foundCommandlineStart = false;
+        for (auto opt : opts)
         {
-            if (arg.find(" ") != std::string::npos)
+            if (opt == subcommand.commandlineOption)
             {
-                buffer += "\"";
-                buffer += arg;
-                buffer += "\"";
+                foundCommandlineStart = true;
             }
-            else
+            else if (foundCommandlineStart)
             {
-                buffer += arg;
+                opt->clear();
             }
-            if (i + 1 < _commandline.size())
+            // otherwise, this option preceeded the start of the commandline
+        }
+
+        // Concatenate all the strings starting with the first arg in the
+        // _commandline as the _real_ commandline.
+        const auto& firstCmdlineArg = _commandline.at(0);
+        auto foundFirstArg = false;
+        std::string fullCommandlineBuffer;
+        for (const auto& arg : _currentCommandline->Args())
+        {
+            if (arg == firstCmdlineArg)
+            {
+                foundFirstArg = true;
+            }
+            if (foundFirstArg)
             {
-                buffer += " ";
+                if (arg.find(" ") != std::string::npos)
+                {
+                    fullCommandlineBuffer += "\"";
+                    fullCommandlineBuffer += arg;
+                    fullCommandlineBuffer += "\"";
+                }
+                else
+                {
+                    fullCommandlineBuffer += arg;
+                }
+                fullCommandlineBuffer += " ";

```

Comment:
big dislike. please file a bug on CLI11 to request a mode that lets us get everything after the first non-parameter string verbatim in the right format.